data-prepper-logstash-configuration/src/main/java/com/amazon/dataprepper/logstash/model/LogstashAttribute.java

```diff
@@ -6,22 +6,38 @@
  * @since 1.2
  */
 public class LogstashAttribute {
-    private String attributeName;
-    private LogstashAttributeValue attributeValue;
+    private final String attributeName;
+    private final LogstashAttributeValue attributeValue;
 
     public String getAttributeName() {
         return attributeName;
     }
 
-    public void setAttributeName(String attributeName) {
-        this.attributeName = attributeName;
-    }
-
     public LogstashAttributeValue getAttributeValue() {
         return attributeValue;
     }
 
-    public void setAttributeValue(LogstashAttributeValue attributeValue) {
-        this.attributeValue = attributeValue;
+    private LogstashAttribute (LogstashAttributeBuilder builder) {
+        this.attributeName = builder.attributeName;
+        this.attributeValue = builder.attributeValue;
+    }
+
+    public static class LogstashAttributeBuilder {
```

Comment:
Let's just call this `Builder`. Otherwise, we will invoke the builder like so: `LogstashAttribute = new LogstashAttribute.LogstashAttributeBuilder().build();`

I would also recommend creating a public static method to get the builder so we can avoid having to use `new`:
```
public static Builder builder() {
    return new Builder();
}
```

Then we could just invoke the builder like so:
```
final LogstashAttribute logstashAttribute = LogstashAttribute.builder()
        .withAttributeValue(attributeValue)
        .withAttributeName("foobar")
        .build();
```